source/docs/v3/tenancy-bootstrappers.blade.md

```diff
@@ -29,31 +29,74 @@ php artisan cache:clear --tag=tenant_123
+
+Since `storage_path()` will be suffixed, your folder structure will look like this:
+
+![The folder structure](/assets/images/file_structure_tenancy.png)
+
+Logs will be saved in `storage/logs` regardless of any changes to `storage_path()` and regardless of the tenant.
+
+### Storage facade
+
+The bootstrapper also makes the `Storage` facade tenant-aware by suffixing the roots of disks listed in `config('tenancy.filesystem.disks')` and by overriding the disk roots in `config('tenancy.filesystem.root_override')` (disk root = the disk path used by the `Storage` facade).
+
+The root of each disk listed in `config('tenancy.filesystem.disks')` will be suffixed. Doing that alone could cause unwanted behavior since Laravel does its own suffixing, so the filesystem config has the `root_override` section, which lets you override the disk roots **after** tenancy has been initialized:
 
-- Suffixes roots of disks used by the `Storage` facade
-- Suffixes `storage_path()` (useful if you're using the local disk for storing tenant files)
-- Makes `asset()` calls use the TenantAssetController to retrieve tenant-specific files
-    - Note: For some assets, e.g. images, you may want to use `global_asset()` (if the asset is shared for all tenants). And for JS/CSS assets, you should use `mix()` or again `global_asset()`.
-    - Because the `asset()` helper uses the TenantAssetController to retrieve tenant-specific files. Be aware this requires the tenant to be identified and initialized to function. Accordingly, you should ensure that the tenant identification middleware used is appropriate for your use case (defaults to InitializeTenancyByDomain). For example, if you are only using subdomain identification, you should update the middleware used by the TenantAssetController to the InitializeTenancyBySubdomain middleware. To do this, you can follow the steps outlined in [configuration]({{ $page->link('configuration') }}) to update the public static property which defined the middleware used, e.g. in your TenancyServiceProvider:
-    
 ```php
-use Stancl\Tenancy\Controllers\TenantAssetsController;
-use Stancl\Tenancy\Middleware\InitializeTenancyByDomainOrSubdomain;
+// Tenancy config (tenancy.filesystem.root_override)
+// %storage_path% gets replaced by storage_path()'s output
+// E.g. Storage::disk('local')->path('') will return "/$path_to_your_application/storage/tenant42/app"
+// (Given a suffix_base of 'tenant' and a tenant with an ID of 42. Same as in the example above in the Storage path helper section)
+'root_override' => [
+    'local' => '%storage_path%/app/',
+    'public' => '%storage_path%/app/public/',
+],
+```
 
-    // other methods in the TenancyServiceProvider
+To make the tenant-aware `Storage` facade work with a custom disk, add the disk's name to `config('tenancy.filesystem.disks')` and override its root in `config('tenancy.filesystem.root_override')` as shown above.
```

Comment:
Perhaps it's worth mentioning that root_overrides are mostly only relevant for local disks and not S3 (not 100% sure but I believe that's the case).

You could add a short paragraph saying that for S3, this is not necessary, and then include an example showing that `Storage::disk('s3')->path('foo.txt')` will return `/tenant42/foo.txt`

Reply:
Updated the paragraph https://github.com/stancl/tenancy-docs/pull/201/commits/9409f28393d31ff101b874bac725024cf732bb3d